linux/src/file_util_linux.c

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#include <stddef.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <inttypes.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include "macro_utils/macro_utils.h"
+
+#include "c_pal/gballoc_hl.h" // IWYU pragma: keep
+#include "c_pal/gballoc_hl_redirect.h" // IWYU pragma: keep
+
+#include "c_logging/logger.h"
+
+#include "c_pal/windows_defines.h"
+
+#include "c_pal/file_util.h"
+
+typedef struct CREATE_FILE_LINUX_TAG
+{
+    int h_file;
+
+} CREATE_FILE_LINUX;
+
+/*Codes_SRS_FILE_UTIL_LINUX_09_002: [ file_util_open_file shall allocate memory for the file handle. ]*/
+HANDLE file_util_open_file(const char* full_file_name, uint32_t access, uint32_t share_mode, LPSECURITY_ATTRIBUTES security_attributes, 
+                    uint32_t creation_disposition, uint32_t flags_and_attributes, HANDLE template_file)
+{
+    (void)flags_and_attributes;
+    
+    CREATE_FILE_LINUX* result;
+    if((full_file_name == NULL) || (full_file_name[0] == '\0'))
+    {
+        /*Codes_SRS_FILE_UTIL_LINUX_09_001: [ If the full_file_name input is either empty or NULL, file_util_open_file shall return an INVALID_HANDLE_VALUE. ]*/
+        LogError("Invalid arguments to file_util_open_file: full_file_name = %s, uint32_t access = %p, uint32_t share_mode = %p, LPSECURITY_ATTRIBUTES security_attributes = %p, uint32_t creation_disposition = %p, uint32_t flags_and_attributes = %p, HANDLE template_file = %p",
+                    full_file_name, access, share_mode, security_attributes, creation_disposition, flags_and_attributes, template_file);
+        result = INVALID_HANDLE_VALUE;
+    } 
+    else 
+    {
+        result = malloc(sizeof(CREATE_FILE_LINUX));
+        if(result == NULL)
+        {
+            result = INVALID_HANDLE_VALUE;
+            LogError("Failure in malloc");
+            
+        }
```

Comment:
Should have requirement for "if there are any failures, yadayada"

Log goes before result assignment.